File: redsheet-portal/src/app/database-models/culture.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { Country, ProjectCulture,  } from './'
export class Culture {
    constructor(culture?: Culture )  {
        let properties = [ "cultureCategoryId", "cultureId", "isEgalitarian", "isIndividualistic", "isMonochronic", "isShortTerm", "countryId", "projectCultureId",];
        for (let property of properties)
            if (culture && culture[property])
                this[property] = culture[property];
    }
 
    @range(1,2147483647)
	cultureCategoryId : number =   undefined;
 
	cultureId : number =   0 ;
 
	isEgalitarian : boolean = false ;
 
	isIndividualistic : boolean = false ;
 
	isMonochronic : boolean = false ;
 
	isShortTerm : boolean = false ;
 
    @range(0,2147483647)
	countryId : number =   undefined;
	country : Country  ;
 
    @range(0,2147483647)
	projectCultureId : number =   undefined;
	projectCulture : ProjectCulture  ;


}

File: RedSheet.DbEntities/Models/Culture.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("Cultures", Schema = "dbo")]
	[Serializable]
	public partial class Culture  
	{
 
	[Range(1, int.MaxValue)]
	public int CultureCategoryId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int CultureId { get; set; } 
 
	[Required]
	public bool IsEgalitarian { get; set; } 
 
	[Required]
	public bool IsIndividualistic { get; set; } 
 
	[Required]
	public bool IsMonochronic { get; set; } 
 
	[Required]
	public bool IsShortTerm { get; set; } 
 
	[RelationshipTableAttribue("Countries","dbo")]
	//Relationships
	public int CountryId { get; set; } 
 
	[ForeignKey("CountryId")]
	public Country Country { get; set; } 
 
	[RelationshipTableAttribue("ProjectCultures","dbo")]
	//Relationships
	public int ProjectCultureId { get; set; } 
 
	[ForeignKey("ProjectCultureId")]
	public ProjectCulture ProjectCulture { get; set; } 

	public Culture()
	{
    }

    }
}

